### PR TITLE
Extend cut_release GHA workflow with release revert mechanism

### PR DESCRIPTION
**Pull request by ytsarev**
* Delete tag and release on associated revert commit
* Conditional expression fix/simplification according to
  https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idif

Signed-off-by: Yury Tsarev <yury.tsarev@absa.africa>

### CODE REVIEW

**Review by jkremser — APPROVED** [review]
looks good, but might be dangerous if someone deletes the tag manually and then also does the revert. That would delete the last but one release.

We might be paranoid and check if the version we are downgrading into in the Chart.yaml == git describe --abbrev=0 --tags ${tag}^ (previous tag in git)

Probably overkill, though + we should be able to catch this in the PR review

**Comment by ytsarev** [comment]
@jkremser really good point, let me try to amend

**Comment by ytsarev** [comment]
@jkremser amended, please check

**Review by k0da — DISMISSED** [review]

File: .github/workflows/cut_release.yaml [hunk 1]
```diff
@@ -12,17 +12,39 @@ jobs:
...
+        run: |
+          tag=$(git describe --tags --abbrev=0)
+          echo "Version to revert: ${tag}"
+          echo "::set-output name=tag::${tag}"
+      - name: Get Previous Tag
+        if: startsWith(github.event.head_commit.message, 'Revert "RELEASE:')
+        id: get_previous_tag
+        run: |
+          tag=$(git describe --tags --abbrev=0 $(git describe --tags --abbrev=0)^)
+          echo "Previos tag: ${tag}"
```
Comment (jkremser) [inline .github/workflows/cut_release.yaml:43]: nit: Previos

File: .github/workflows/cut_release.yaml [hunk 2]
```diff
@@ -12,17 +12,39 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v2
-      - name: Get Tag
-        id: get_tag
+      - name: Get New Tag
```
Comment (jkremser) [inline .github/workflows/cut_release.yaml:17]: (nit) now that we can travel both: to the future and to the past, perhaps "desired tag" would better capture the intent

**Review by jkremser — CHANGES_REQUESTED** [review]

**Comment by ytsarev** [comment]
@jkremser make sense, amended

**Review by jkremser — APPROVED** [review]

**Review by k0da — APPROVED** [review]

**Review by kuritka — APPROVED** [review]

**Pull request merged**